### PR TITLE
Added the scorecard github action and its badge

### LINKED ISSUE #14404: Enable OpenSSF Scorecard to enhance security practices across the project
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**
Hi, I'm Harshita. I’m working with [CNCF and the Google Open Source Security Team for the GSoC 2024 term](https://github.com/cncf/mentoring/issues/1196). We are collaborating to enhance security practices across various CNCF projects. The goal is to improve security for all CNCF projects by both using OpenSSF Scorecards and implementing its security improvements.

The [Open Source Security Foundation](https://openssf.org/) (OpenSSF) [Scorecard](https://github.com/ossf/scorecard) is a tool designed to evaluate the security posture of open-source projects. This has the [Scorecard GitHub Action](https://github.com/ossf/scorecard-action), which automates the process by running security checks on the GitHub repository. By integrating this Action into the repository's workflow, developers can continuously monitor the project’s security posture. The Scorecard checks cover various security best practices and provide scores for multiple categories. Some checks include Code Reviews, Branch Protection, Signed Releases, etc.

The workflow runs on every change in the main branch. It publishes the Scorecard checks' results to the project's security dashboard and includes suggestions on how to solve any issues. This Action has already been adopted by 1800+ projects, with prominent users like [Tensorflow](https://github.com/tensorflow/tensorflow), [Angular](https://github.com/angular/angular), [sos.dev](http://sos.dev/), [deps.dev](http://deps.dev/), and many CNCF projects.

Once the Scorecard GitHub Action is set up and running, the results can be displayed as a badge in the repository's README file. This badge serves as a quick indicator of the project's security posture, helping users and contributors evaluate the project's security practices quickly.

**What is use case behind this feature:**

The OpenSSF Scorecard improves open-source project's security by providing automated, transparent assessments of their security practices. It will help you identify vulnerabilities, adhere to best practices, and continuously enhance your security posture, increasing user trust and reducing the risk of security exploits.

I'll be the one to create the PR to add the scorecard GitHub action, and I will also work with you to remediate the identified vulnerabilities. I'll go through each [scorecard check](https://github.com/ossf/scorecard/blob/main/docs/checks.md) to see where the score has dropped and how it can be improved.

Would you be interested in a PR which adds this Action?

/cc @joycebrum @diogoteles08 @pnacht @nate-double-u

### PR DESCRIPTION
**Pull request by harshitasao**
Resolves #14404 

**Checklist:**

- [X] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [X] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [X] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by subhamkrai — CHANGES_REQUESTED** [review]
@harshitasao please fix the commitlint CI.

File: .github/workflows/scorecards.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736 # v2.3.1
```
Comment (subhamkrai) [inline .github/workflows/scorecards.yml:36]:
```suggestion
        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736 # v2.3.1
```

same?

File: .github/workflows/scorecards.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
Comment (subhamkrai) [inline .github/workflows/scorecards.yml:31]:
```suggestion
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

can we have some version?

**Review by subhamkrai — CHANGES_REQUESTED** [review]

**Comment by travisn** [comment]
Before we merge this, let's take a quick look at the report to see what low hanging fruit there is to improve the score. Thanks @obnoxxx for taking a look!

**Comment by harshitasao** [comment]
@travisn  As my next step, I will be creating a separate scorecard score improvement issue [something like this](https://github.com/argoproj/argo-workflows/issues/12031) and fix each check where the score is dropping, thus improving the project's security posture.

**Comment by BlaineEXE** [comment]
Hi @harshitasao, we have addressed one of the high priority items. Could you re-run the workflow so we can see what the latest score looks like? Alternatively, it would be nice to know what steps you used to run the workflow and get a result published.

File: .github/workflows/scorecards.yml (hunk 2 above)
Comment (travisn) [reply .github/workflows/scorecards.yml:31]: @harshitasao Could you answer this?

**Comment by harshitasao** [comment]
> Hi @harshitasao, we have addressed one of the high priority items. Could you re-run the workflow so we can see what the latest score looks like? Alternatively, it would be nice to know what steps you used to run the workflow and get a result published.

We cannot run the action manually; it will run only after getting merged into the project. If the action is not added to the project, the Scorecard team runs a weekly scan of public GitHub repositories in order to track the overall security health of the open source ecosystem.

File: .github/workflows/scorecards.yml (hunk 2 above)
Comment (obnoxxx) [reply .github/workflows/scorecards.yml:31]: This  sounds like a plausible justification to me @subhamkrai , @travisn

**Comment by obnoxxx** [comment]
@subhamkrai  wrote:
> @harshitasao please fix the commitlint CI.

This has been fixed meanwhile 👍🏼

**Review by obnoxxx — APPROVED** [review]
Thanks for the updates, @harshitasao !
I think requests have been addressed and explained.
LGTM now.

**Review by travisn — APPROVED** [review]

**Pull request merged**